Fails on darwin https://github.com/NixOS/nixpkgs/pull/67383/checks?check_run_id=202217794

```
[33/38] Compiling C object 'app/a172ced@@test_queue@exe/tests_test_queue.c.o'.
FAILED: app/a172ced@@test_queue@exe/tests_test_queue.c.o
/nix/store/4yhnj6g8ixx31jgil477xrir5y3n39im-clang-wrapper-7.1.0/bin/cc -Iapp/a172ced@@test_queue@exe -Iapp -I../app -I../app/src -I/nix/store/gl5yaaflmwpap4sxnbjjck243lpzcpys-ffmpeg-3.4.6-dev/include -I/nix/store/0pn4dchsfz57xi5b3dnwci44krnd53g2-SDL2-2.0.9-dev/include/SDL2 -Xclang -fcolor-diagnostics -pipe -std=c11 -D_THREAD_SAFE -MD -MQ 'app/a172ced@@test_queue@exe/tests_test_queue.c.o' -MF 'app/a172ced@@test_queue@exe/tests_test_queue.c.o.d' -o 'app/a172ced@@test_queue@exe/tests_test_queue.c.o' -c ../app/tests/test_queue.c
../app/tests/test_queue.c:12:5: error: use of undeclared identifier 'NULL'
    queue_init(&queue);
    ^
../app/src/queue.h:16:40: note: expanded from macro 'queue_init'
    (void) ((PQ)->first = (PQ)->last = NULL)
                                       ^
../app/tests/test_queue.c:19:5: error: use of undeclared identifier 'NULL'
    queue_push(&queue, next, &v1);
    ^
../app/src/queue.h:52:29: note: expanded from macro 'queue_push'
        (ITEM)->NEXTFIELD = NULL; \
                            ^
../app/tests/test_queue.c:20:5: error: use of undeclared identifier 'NULL'
    queue_push(&queue, next, &v2);
    ^
../app/src/queue.h:52:29: note: expanded from macro 'queue_push'
        (ITEM)->NEXTFIELD = NULL; \
                            ^
3 errors generated.
ninja: build stopped: subcommand failed.
```

--- a/pkgs/misc/scrcpy/default.nix
+++ b/pkgs/misc/scrcpy/default.nix
@@ -2,6 +2,7 @@
 , meson
 , ninja
 , pkgconfig
+, fetchpatch
 
 , platform-tools
 , ffmpeg
@@ -9,10 +10,10 @@
 }:
 
 let
-  version = "1.8";
+  version = "1.10";
   prebuilt_server = fetchurl {
     url = "https://github.com/Genymobile/scrcpy/releases/download/v${version}/scrcpy-server-v${version}.jar";
-    sha256 = "1h755k5xpchlm7wq2yk5mlwjnh7y4yhviffixacby0srj3pmb443";
+    sha256 = "144k25x6ha89l9p5a1dm6r3fqvgqszzwrhvkvk0r44vg0i71msyb";
   };
 in
 stdenv.mkDerivation rec {
@@ -23,7 +24,7 @@ stdenv.mkDerivation rec {
     owner = "Genymobile";
     repo = pname;
     rev = "v${version}";
-    sha256 = "1cx7y3w699s3i8s53l1mb7lkrnbix457hf17liwh00jzb0i7aga7";
+    sha256 = "0hhncqcs49n9g8sgvwbyvkaq4b1dhrpn7qgnaj6grjcb0i27vzaq";
   };
 
   # postPatch:
@@ -39,12 +40,18 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ ffmpeg SDL2 ];
 
+  # FIXME: remove on update to > 1.10
+  patches = [(fetchpatch {
+    url = "https://github.com/Genymobile/scrcpy/commit/c05056343b56be65ae887f8b7ead61a8072622b9.diff";
+    sha256 = "1xh24gr2g2i9rk0zyv19jx54hswrq12ssp227vxbhsbamin9ir5b";
+  })];
+
   # Manually install the server jar to prevent Meson from "fixing" it
   preConfigure = ''
     echo -n > server/meson.build
   '';
 
-  mesonFlags = ["-Doverride_server_path=${prebuilt_server}"];
+  mesonFlags = [ "-Doverride_server_path=${prebuilt_server}" ];
   postInstall = ''
     mkdir -p "$out/share/scrcpy"
     ln -s "${prebuilt_server}" "$out/share/scrcpy/scrcpy-server.jar"

